src/io/filewatcher.rs

```diff
@@ -53,17 +56,33 @@ impl FileWatcher {
         // Reset if file has been remade (i.e. is shorter) and update state
         if (meta.len() as usize) < self.last_size {
```

Comment:
I think it would be worth adding some logging, something like a `tracing::warn!("File has shortened, the file may have been replaced. Reopening.");` would be good.